src/lint/collect-spelling-diagnostics.ts

```diff
@@ -0,0 +1,60 @@
+import type { LintingError } from './algorithm-error-reporter-type';
+
+import { offsetToLineAndColumn } from './utils';
+
+let ruleId = 'spelling';
+
+// Note that these will be composed, so cannot contain backreferences
+let matchers = [
+  {
+    pattern: /\*this\* object/giu,
+    message: 'Prefer "the *this* value"',
+  },
+  {
+    pattern: /1's complement/giu,
+    message: 'Prefer "one\'s complement"',
+  },
+  {
+    pattern: /2's complement/giu,
+    message: 'Prefer "two\'s complement"',
+  },
+  {
+    pattern: /\*0\*/gu,
+    message: 'The Number value 0 should be written "*+0*", to unambiguously exclude "*-0*"',
+  },
+  {
+    pattern: /behavior/giu,
+    message: 'ECMA-262 uses UK spelling ("behaviour")',
```

Comment:
```suggestion
    message: 'ECMA-262 uses Oxford spelling ("behaviour")',
```
nit: Probably worth being [precise](https://en.wikipedia.org/wiki/Oxford_spelling).

Reply:
Done, thanks.